e2e/backend_service_update_test.go

```diff
@@ -226,3 +255,86 @@ func TestBackendServiceUpdate(t *testing.T) {
 	wantBS = &compute.BackendService{LoadBalancingScheme: "INTERNAL_MANAGED"}
 	checkBackendService(t, ctx, theCloud, bs1ID, wantBS, compareLBScheme)
 }
+
+func TestBackendServiceMetadataUpdate(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	graphBuilder := rgraph.NewBuilder()
+	resUniqueIdPart := "test-bs-metadata-update"
+
+	hcName := resourceName(resUniqueIdPart + "-hc")
+	hcID, err := buildHealthCheck(graphBuilder, hcName, 15)
+	if err != nil {
+		t.Fatalf("buildHealthCheck(_, %s, _) = (_, %v) want (_, nil)", hcName, err)
+	}
+
+	bsName := resourceName(resUniqueIdPart + "-bs")
+	bsID, err := buildBackendServiceWithMetadata(graphBuilder, bsName, hcID, map[string]string{"foo": "1", "bar": "2", "baz": "2"})
```

Comment:
use wantedMetadata variable (you need to move it up:))

Reply:
Done.